### PR TITLE
added the new url format for Gronkh and tests [Issue #5531]

### LINKED ISSUE #5531: Gronkh.tv Unsupported URL, new URL not recognized
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.11.11** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

_No response_

### Provide a description that is worded well enough to be understood

Gronkh.tv seems to have changed a part of the URL for streams from /stream/ to /streams/ (plural now).
The old URLs are still supported, as the test URL redirects to the new URL. But calling yt-dlp with the new URL is raising an "Unsupported URL"-error, because yt-dlp is not recognizing the new url.

I confirmed it as source of the error by changing  _VALID_URL in the extractor yt_dlp/extractor/gronkh.py, after which it worked fine. I don't know whether both URLs will stay valid, or just the plural-version, as there seems to be still much work done on the site, so maybe support both?

Old URL: https://gronkh.tv/stream/536
New URL: https://gronkh.tv/streams/536

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['--format', 'bestvideo[height<=720]+bestaudio/best[height<=720]', '--restrict-filenames', '--output', '[%(extractor)s][%(channel)s] %(title)s [%(id)s].%(ext)s', '--no-overwrites', '--no-playlist', '--all-subs', '--embed-subs', '-vU', '--merge-output-format', 'mkv', 'https://gronkh.tv/streams/536']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.11.11 [8b644025b] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: 692e9ccbe
[debug] Python 3.10.8 (CPython x86_64 64bit) - Linux-6.0.0-2-amd64-x86_64-with-glibc2.35 (OpenSSL 3.0.5 5 Jul 2022, glibc 2.35)
[debug] exe versions: ffmpeg 5.1.2 (fdk,setts), ffprobe 5.1.2
[debug] Optional libraries: Cryptodome-3.11.0, brotli-1.0.9, certifi-2022.06.15, pyxattr-0.7.2, secretstorage-3.3.3, sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1725 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.11.11, Current version: 2022.11.11
yt-dlp is up to date (2022.11.11)
[debug] [generic] Extracting URL: https://gronkh.tv/streams/536
[generic] 536: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] 536: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://gronkh.tv/streams/536
Traceback (most recent call last):
  File "/path/to/yt-dlp/yt_dlp/YoutubeDL.py", line 1493, in wrapper
    return func(self, *args, **kwargs)
  File "/path/to/yt-dlp/yt_dlp/YoutubeDL.py", line 1569, in __extract_info
    ie_result = ie.extract(url)
  File "/path/to/yt-dlp/yt_dlp/extractor/common.py", line 674, in extract
    ie_result = self._real_extract(url)
  File "/path/to/yt-dlp/yt_dlp/extractor/generic.py", line 2721, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://gronkh.tv/streams/536
```

### PR DESCRIPTION
**Pull request by muddi900**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

Gonkh.tv switched to a new url format. I have updated the regex to include both the new scheme and the old one. I have added new tests. However the tests will fail because `view_count` field is required and it is a dynamic value. I have tested it with the current accurate value and it works.

Fixes #5531

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

**Comment by bashonly** [comment]
Don't overwrite the old test, keep the old and add the new

> However the tests will fail because view_count field is required and it is a dynamic value

```py
'view_count': int,
```

**Comment by muddi900** [comment]
> Don't overwrite the old test case, keep the old and add the new. We want tests for both URL formats
> 
> > However the tests will fail because view_count field is required and it is a dynamic value
> 
> ```python
> 'view_count': int,
> ```

Thanks. The tests are passing

File: yt_dlp/extractor/gronkh.py [hunk 1]
```diff
@@ -9,23 +9,39 @@
...
+            'view_count': int,
+            'thumbnail': 'https://01.cdn.vod.farm/preview/9e2555d3a23bf4e5c5b7c6b3b70a9d84.jpg',
+            'upload_date': '20221111'
+        },
+        'params': {'skip_download': True}
+    }, {
+        'url': 'https://gronkh.tv/streams/657',
+        'only_matching': True,
+    },
+        {
```
Comment (bashonly) [inline yt_dlp/extractor/gronkh.py:29]: this `only_matching` test case is redundant
```suggestion
```

File: yt_dlp/extractor/gronkh.py [hunk 2]
```diff
@@ -9,23 +9,39 @@
...
             'thumbnail': 'https://01.cdn.vod.farm/preview/6436746cce14e25f751260a692872b9b.jpg',
             'upload_date': '20211001'
         },
         'params': {'skip_download': True}
     }, {
         'url': 'https://gronkh.tv/watch/stream/546',
         'only_matching': True,
-    }]
+    }
+    ]
```
Comment (bashonly) [inline yt_dlp/extractor/gronkh.py:44]:
```suggestion
    }]
```

**Review by bashonly — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/gronkh.py [hunk 3]
```diff
@@ -9,23 +9,33 @@
...
+            'id': '657',
+            'ext': 'mp4',
+            'title': 'H.O.R.D.E. - DAS ZWEiTE ZEiTALTER 🎲 Session 1',
+            'view_count': int,
+            'thumbnail': 'https://01.cdn.vod.farm/preview/9e2555d3a23bf4e5c5b7c6b3b70a9d84.jpg',
+            'upload_date': '20221111'
+        },
+        'params': {'skip_download': True}
+    },
+        {
```
Comment (pukkandan) [inline yt_dlp/extractor/gronkh.py:26]:
```suggestion
    }, {
```

File: yt_dlp/extractor/gronkh.py [hunk 4]
```diff
@@ -9,23 +9,33 @@
...
             'thumbnail': 'https://01.cdn.vod.farm/preview/6436746cce14e25f751260a692872b9b.jpg',
             'upload_date': '20211001'
         },
         'params': {'skip_download': True}
-    }, {
-        'url': 'https://gronkh.tv/watch/stream/546',
-        'only_matching': True,
-    }]
+    },
+    ]
```
Comment (pukkandan) [inline yt_dlp/extractor/gronkh.py:38]:
```suggestion
    }]
```

**Pull request merged**